specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/deviceSecurityGroups.json

```diff
@@ -0,0 +1,562 @@
+        }
+      ],
+      "properties": {
+        "allowlistValues": {
+          "type": "array",
+          "description": "The values to allow. The format of the values depends on the rule type.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "ruleType": {
+          "type": "string",
+          "description": "The type of the custom alert rule.",
+          "enum": [
+            "ConnectionToIpNotAllowed",
+            "LocalUserNotAllowed",
+            "ProcessNotAllowed"
+          ],
+          "x-ms-enum": {
+            "name": "ruleType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "ConnectionToIpNotAllowed",
+                "description": "Outbound connection to an ip that isn't allowed. Allow list consists of ipv4 or ipv6 range in CIDR notation."
+              },
+              {
+                "value": "LocalUserNotAllowed",
+                "description": "Login by a local user that isn't allowed. Allow list consists of login names to allow."
+              },
+              {
+                "value": "ProcessNotAllowed",
+                "description": "Execution of a process that isn't allowed. Allow list consists of process names to allow."
+              }
+            ]
+          }
+        }
+      },
+      "required": [
+        "allowlistValues"
+      ]
+    },
+    "DenylistCustomAlertRule": {
+      "type": "object",
+      "description": "A custom alert rule that checks if a value (depends on the custom alert type) is denied.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ListCustomAlertRule"
+        }
+      ],
+      "properties": {
+        "denylistValues": {
+          "type": "array",
+          "description": "The values to deny. The format of the values depends on the rule type.",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "required": [
+        "denylistValues"
+      ]
+    },
+    "ThresholdCustomAlertRule": {
+      "type": "object",
+      "description": "A custom alert rule that checks if a value (depends on the custom alert type) is within the given range.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CustomAlertRule"
+        }
+      ],
+      "properties": {
+        "minThreshold": {
+          "type": "integer",
+          "description": "The minimum threshold."
+        },
+        "maxThreshold": {
+          "type": "integer",
+          "description": "The maximum threshold."
+        }
+      },
+      "required": [
+        "minThreshold",
+        "maxThreshold"
+      ]
+    },
+    "TimeWindowCustomAlertRule": {
+      "type": "object",
+      "description": "A custom alert rule that checks if the number of activities (depends on the custom alert type) in a time window is within the given range.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CustomAlertRule"
```

Comment:
ThresholdCustomAlertRule contains all CustomAlertRule's definition. So why do we need both of them?

Reply:
Good catch, Thanks